tests/test_timedelta_isoformat.py

```diff
@@ -112,6 +112,8 @@
     # scientific notation in designated values
     ("P1.0e+1D", "unexpected character 'e'"),
     ("P10.0E-1D", "unexpected character 'E'"),
+    # attempt to cause the parser to confuse duration tokens and timedelta arguments
+    ("P1years1M", "unexpected character 'y'"),
```

Comment:
I'd like to merge this test case while we continue to consider the other changes.

Are you OK with me extracting it into a separate commit for you?  (I'll add you as the author in the `git` commit entry)

Reply:
(this is safe to ignore now; the pull request has been merged)